<launch>
    <machine name="local" address="localhost" default="true"/>
    <node name="Chisel" pkg="chisel_ros" type="ChiselNode" output="screen" launch-prefix="taskset -c 0,1">
    <!--node name="Chisel" pkg="chisel_ros" type="ChiselNode" output="log"/-->
        <param name="chunk_size_x" value="8"/>  <!-- 8 -->
        <param name="chunk_size_y" value="8"/>
        <param name="chunk_size_z" value="8"/>
        <param name="truncation_scale" value="2.0"/>  <!-- 2 -->
        <param name="integration_weight" value="1.0"/>
        <param name="use_voxel_carving" value="true"/>
        <param name="use_color" value="true"/>
        <param name="carving_dist_m" value="0.01"/>
        <param name="voxel_resolution_m" value="0.2"/>  <!-- 0.1 for demo, 0.2 for bbgf -->
        <param name="far_plane_dist" value="6.0"/>
        <param name="near_plane_dist" value="0.04"/>
        <param name="base_transform" value="/world"/>
        <param name="transform_name" value="/ref_frame"/>
        <param name="fusion_mode" value="DepthImage"/>

        <remap from="/odom_topic" to="/vins_estimator/camera_pose"/>
        <remap from="/depth_image" to="/sgbm_ros_node/depth_image"/>
        <remap from="/depth_camera_info" to="/stereo/rect/camera_info"/>
        <remap from="/color_image" to="/stereo/left/image_rect"/>
        <remap from="/color_camera_info" to="/stereo/left/camera_info"/>

        <param name="calc_mesh" value="true"/>
        <param name="three_camera_mode" value="false"/>
    </node>
</launch>
